src/menu-button.js

```diff
@@ -0,0 +1,59 @@
+'use strict';
+
+const React = require('react');
+const classnames = require('classnames');
+const Icon = require ('./icon');
+
+const baseClasses = {
+  'mdl-button': true,
+  'mdl-js-button': true,
+  'mdl-button--icon': true
+};
+
+class MenuButton extends React.Component {
+  constructor(...args){
+    super(...args);
+    this._autoId = '_' + Math.random().toString(36).slice(2);
+
+  }
+  render() {
+    const {
+      className,
+      children
+    } = this.props;
+
+    const classes = classnames(baseClasses, className);
+
+    let {
+      id
+    } = this.props;
+
+    if(!id) {
+      id = this._autoId;
+    }
+
+    let defaultIcon;
+    if(children){
+      defaultIcon = (
+        <Icon name={children}/>
```

Comment:
children should be passed directly, they shouldn't be placed as the name of an icon